### PR TITLE
Replace custom client test code with mockery generated one

### PR DESCRIPTION
**Pull request by gkrenn**
## Description

This PR replaces the client mock structs we used with mockery generated ones.

In addition the gci make command was updated to not change imports for generated files.

## How can this be tested?

Run unit tests and check if everything is still working.

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: pkg/controllers/csi/provisioner/controller_test.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/connectioninfo"
 	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/dynatraceclient"
 	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer"
+	mockedclient "github.com/Dynatrace/dynatrace-operator/test/mocks/pkg/clients/dynatrace"
```
Comment (andriisoldatenko) [inline pkg/controllers/csi/provisioner/controller_test.go:17]:
```suggestion
	mock "github.com/Dynatrace/dynatrace-operator/test/mocks/pkg/clients/dynatrace"
```

File: pkg/controllers/csi/provisioner/controller_test.go [hunk 2]
```diff
@@ -190,7 +191,7 @@ func TestOneAgentProvisioner_Reconcile(t *testing.T) { //nolint:revive
 				},
 			},
 		)
-		mockClient := &dtclient.MockDynatraceClient{}
+		mockClient := &mockedclient.Client{}
```
Comment (andriisoldatenko) [inline pkg/controllers/csi/provisioner/controller_test.go:194]:
```suggestion
		mock := &mockedclient.Client{}
```
since it already has name `Client`.

**Review by andriisoldatenko — CHANGES_REQUESTED** [review]
please apply same change for all imports/usage.

File: pkg/controllers/csi/provisioner/controller_test.go [hunk 3]
```diff
@@ -301,7 +302,7 @@ func TestOneAgentProvisioner_Reconcile(t *testing.T) { //nolint:revive
 		errorfs := &mkDirAllErrorFs{
 			Fs: afero.NewMemMapFs(),
 		}
-		mockClient := &dtclient.MockDynatraceClient{}
+		mockClient := &mockedclient.Client{}
```
Comment (aorcholski) [inline pkg/controllers/csi/provisioner/controller_test.go:305]: IMHO we should create mocked client using NewClient(t) if any expectations are defined (by `On(..)`). Then the library checks if expectations are correct or not

File: pkg/controllers/dynakube/activegate/internal/authtoken/reconciler_test.go [hunk 4]
```diff
@@ -41,7 +42,7 @@ func newTestReconcilerWithInstance(client client.Client) *Reconciler {
 			APIURL: "https://testing.dev.dynatracelabs.com/api",
 		},
 	}
-	dtc := &dtclient.MockDynatraceClient{}
+	dtc := &mockedclient.Client{}
```
Comment (aorcholski) [inline pkg/controllers/dynakube/activegate/internal/authtoken/reconciler_test.go:45]: NewClient(t)

File: pkg/controllers/csi/provisioner/controller_test.go [hunk 5]
```diff
@@ -426,7 +427,7 @@ func TestOneAgentProvisioner_Reconcile(t *testing.T) { //nolint:revive
 	t.Run("error getting dynakube from db", func(t *testing.T) {
 		gc := &CSIGarbageCollectorMock{}
 		memFs := afero.NewMemMapFs()
-		mockClient := &dtclient.MockDynatraceClient{}
+		mockClient := &mockedclient.Client{}
```
Comment (aorcholski) [inline pkg/controllers/csi/provisioner/controller_test.go:430]: NewClient(t)

File: pkg/controllers/csi/provisioner/controller_test.go [hunk 6]
```diff
@@ -352,7 +353,7 @@ func TestOneAgentProvisioner_Reconcile(t *testing.T) { //nolint:revive
 		var revision uint = 3
 		testProcessModuleConfig := createTestProcessModuleConfig(revision)
 		memFs := afero.NewMemMapFs()
-		mockClient := &dtclient.MockDynatraceClient{}
+		mockClient := &mockedclient.Client{}
```
Comment (aorcholski) [inline pkg/controllers/csi/provisioner/controller_test.go:356]: NewClient(t)

**Review by aorcholski — CHANGES_REQUESTED** [review]

File: pkg/controllers/csi/provisioner/controller_test.go (hunk 2 above)
Comment (gkrenn) [reply pkg/controllers/csi/provisioner/controller_test.go:194]: Removed the explicit name before, such that the imports name is "mocks" then, but decided for using the "mockedclient" name, as we sometimes import other mock packages and the naming clashes.

But I agree, we could remove the "mockedclient" naming from packages where there are no problems.

File: pkg/controllers/csi/provisioner/controller_test.go (hunk 3 above)
Comment (gkrenn) [reply pkg/controllers/csi/provisioner/controller_test.go]: You are right, thanks for pointing out

**Review by aorcholski — DISMISSED** [review]

**Review by andriisoldatenko — DISMISSED** [review]

**Review by 0sewa0 — APPROVED** [review]

**Review by andriisoldatenko — APPROVED** [review]

**Review by aorcholski — APPROVED** [review]

**Pull request merged**